File: backend/models/employee.py
from sqlalchemy import Column, Integer, String, Numeric, DateTime, Text, Date, Float, ForeignKey
from sqlalchemy.dialects.postgresql import JSONB
from sqlalchemy.sql import func
from database import Base


class Employee(Base):
    """Legacy model – the employees table was merged into users by migration 0013.
    Kept only so Alembic can reference the table during downgrades / history."""
    __tablename__ = "employees"

    id                  = Column(Integer, primary_key=True, index=True)
    company_id          = Column(Integer, ForeignKey("companies.id", ondelete="SET NULL"), nullable=True)
    name                = Column(String(255), nullable=False)
    gender              = Column(String(10), nullable=True)
    date_of_birth       = Column(Date, nullable=True)
    blood_group         = Column(String(5), nullable=True)
    marital_status      = Column(String(20), nullable=True)
    emergency_contact   = Column(String(20), nullable=True)
    emergency_name      = Column(String(255), nullable=True)
    phone               = Column(String(20), nullable=True)
    email               = Column(String(255), nullable=True)
    phone_verified      = Column(String(1), nullable=True, default="N")
    email_verified      = Column(String(1), nullable=True, default="N")
    address             = Column(Text, nullable=False)
    aadhar_number       = Column(String(12), nullable=False, unique=True)
    bank_account_number = Column(String(18), nullable=False)
    ifsc_code           = Column(String(11), nullable=True)
    bank_name           = Column(String(255), nullable=True)
    kyc_status          = Column(String(20), nullable=True, default="pending")
    kyc_verified_name   = Column(String(255), nullable=True)
    hourly_rate         = Column(Numeric(10, 2), nullable=False, default=0.00)
    shift               = Column(String(10), nullable=False, default="SHIFT_A")
    face_encoding       = Column(JSONB, nullable=True)
    photo               = Column(Text, nullable=True)
    work_location_name  = Column(String(255), nullable=True)
    work_latitude       = Column(Float, nullable=True)
    work_longitude      = Column(Float, nullable=True)
    attendance_radius_km = Column(Float, nullable=True, default=10.0)
    created_at          = Column(DateTime(timezone=True), server_default=func.now())
    updated_at          = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

File: mobile/app/src/main/java/com/econetvision/erp/data/model/Employee.kt
package com.econetvision.erp.data.model

import com.google.gson.annotations.SerializedName

data class Employee(
    val id: Int,
    val name: String,
    val gender: String?,
    @SerializedName("date_of_birth") val dateOfBirth: String?,
    @SerializedName("blood_group") val bloodGroup: String?,
    @SerializedName("marital_status") val maritalStatus: String?,
    @SerializedName("emergency_contact") val emergencyContact: String?,
    @SerializedName("emergency_name") val emergencyName: String?,
    val address: String?,
    @SerializedName("aadhar_number") val aadharNumber: String?,
    @SerializedName("bank_account_number") val bankAccountNumber: String?,
    @SerializedName("ifsc_code") val ifscCode: String?,
    @SerializedName("bank_name") val bankName: String?,
    @SerializedName("kyc_status") val kycStatus: String?,
    @SerializedName("hourly_rate") val hourlyRate: Double?,
    val shift: String?,
    val photo: String?,
    @SerializedName("work_location_name") val workLocationName: String?,
    @SerializedName("work_latitude") val workLatitude: Double?,
    @SerializedName("work_longitude") val workLongitude: Double?,
    @SerializedName("attendance_radius_m") val attendanceRadiusM: Double?,
    @SerializedName("created_at") val createdAt: String,
    @SerializedName("updated_at") val updatedAt: String
)

data class PaginatedEmployees(
    val items: List<Employee>,
    val total: Int,
    val page: Int,
    @SerializedName("per_page") val perPage: Int,
    val pages: Int
)
